packages/cli/lib/project-generator.js

```diff
@@ -55,6 +56,24 @@ module.exports = class ProjectGenerator extends BaseGenerator {
       const isValid = utils.validate(this.args[0]);
       if (typeof isValid === 'string') throw new Error(isValid);
     }
+
+    this.setupRenameTransformer();
+  }
+
+  /**
+   * Registers a Transform Stream with Yeoman. Removes `.template` extension
+   * from files that have it during project generation.
+   */
+  setupRenameTransformer() {
+    this.registerTransformStream(
+      rename(function(file) {
+        if (file.extname === '.template') {
```

Comment:
`file.extname` property only gives us access to the last `.extname` in the basename of a file. I can do a check again `file.basename` instead but I think that makes the code messy.

@raymondfeng

Reply:
We can do the following:
```js
const fileName = `${file.basename}.${file.extname}`;
const result = fileName.match(/(.+)\.(ts|js)\.template$/);
if (result) {
  file.extname = result[2];
  file.basename = result[1];
}
```

Reply:
I agree that we shouldn't lean on it being `*.template`, since it means someone could put a non-TS/JS template in there and we would mistakenly rewrite the results as a `.ts` or `.js` file.

Reply:
(I don't know how much we really _should_ worry or care about that, but the cost of not letting others be silly in that way is minimal.)